recipes/flatcc/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+    }
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            if self.settings.compiler == "Visual Studio" and self.options.shared:
+                #Building flatcc shared libs with Visual Studio is broken
+                raise ConanInvalidConfiguration("Building flatcc libraries shared is not supported")
+            if self.settings.compiler == "gcc":
+                raise ConanInvalidConfiguration("Building flatcc with MinGW is not supported")
+        if self.settings.os == "Macos" and self.options.shared:
+            raise ConanInvalidConfiguration("Building flatcc shared on Macos is currently not supported due to Systems Integrity Protection problems")
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["FLATCC_PORTABLE"] = self.options.portable
+            self._cmake.definitions["FLATCC_GNU_POSIX_MEMALIGN"] = self.options.gnu_posix_memalign
+            self._cmake.definitions["FLATCC_RTONLY"] = self.options.runtime_lib_only
+            self._cmake.definitions["FLATCC_INSTALL"] = True
+            self._cmake.definitions["FLATCC_COVERAGE"] = False
+            self._cmake.definitions["FLATCC_DEBUG_VERIFY"] = self.options.verify_assert
+            self._cmake.definitions["FLATCC_TRACE_VERIFY"] = self.options.verify_trace
+            self._cmake.definitions["FLATCC_REFLECTION"] = self.options.reflection
+            self._cmake.definitions["FLATCC_NATIVE_OPTIM"] = self.options.native_optim
+            self._cmake.definitions["FLATCC_FAST_DOUBLE"] = self.options.fast_double
+            self._cmake.definitions["FLATCC_IGNORE_CONST_COND"] = self.options.ignore_const_condition
+            self._cmake.definitions["FLATCC_TEST"] = False
+            #self._cmake.configure(source_folder=os.path.join(self.source_folder, self._source_subfolder))
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        if self.settings.build_type == "Debug" and not tools.os_info.is_windows:
+            debug_suffix = "_d" if self.settings.build_type == "Debug" else ""
```

Comment:
Why do you add debug suffix, if the upstream doesn't? Keep the project as close as possible to the original form.

Reply:
I added the suffix in conanfile.py only because the cmake files of flatcc add them to the libraries. So upstream does that, not me. Except that they don't add it on Windows....

Reply:
thanks for the clarification!